L1Trigger/L1THGCal/src/backend/HGCalHistoSeedingImpl.cc

```diff
@@ -517,56 +498,75 @@ std::vector<std::pair<GlobalPoint, double>> HGCalHistoSeedingImpl::computeSecond
       
       /* smoothen along the r/z direction */
       smoothHistoCluster = fillSmoothRPhiHistoClusters(smoothPhiHistoCluster);
     } else if (seedingSpace_ == XY) {
+
+      navigator_ = Navigator(nBins1_, Navigator::AxisType::Bounded, nBins2_, Navigator::AxisType::Bounded); 
+      
       smoothHistoCluster = fillSmoothHistoClusters(histoCluster, smoothing_ecal_, Bin::Content::Ecal);
       smoothHistoCluster = fillSmoothHistoClusters(smoothHistoCluster, smoothing_hcal_, Bin::Content::Hcal);
       // Update sum with smoothen ECAL + HCAL
       for (int z_side : {-1, 1}) {
-	for (unsigned x1 = 0; x1 < nBins1_; x1++) {
-	  for (unsigned x2 = 0; x2 < nBins2_; x2++) {
-	    auto& bin = smoothHistoCluster.at(z_side, x1, x2);
-	    bin.values[Bin::Content::Sum] = bin.values[Bin::Content::Ecal] + bin.values[Bin::Content::Hcal];
-	  }
-	}
+        for (unsigned x1 = 0; x1 < nBins1_; x1++) {
+          for (unsigned x2 = 0; x2 < nBins2_; x2++) {
+            auto& bin = smoothHistoCluster.at(z_side, x1, x2);
+            bin.values[Bin::Content::Sum] = bin.values[Bin::Content::Ecal] + bin.values[Bin::Content::Hcal];
+          }
+        }
       }
     }
-    
-  /* seeds determined with local maximum criteria */
-  if (seedingType_ == HistoMaxC3d)
-    seedPositionsEnergy = computeMaxSeeds(smoothHistoCluster);
-  else if (seedingType_ == HistoThresholdC3d)
-    seedPositionsEnergy = computeThresholdSeeds(smoothHistoCluster);
-  else if (seedingType_ == HistoInterpolatedMaxC3d)
-    seedPositionsEnergy = computeInterpolatedMaxSeeds(smoothHistoCluster);
-  else if (seedingType_ == HistoSecondaryMaxC3d)
-    seedPositionsEnergy = computeSecondaryMaxSeeds(smoothHistoCluster);
+
+    /* seeds determined with local maximum criteria */
+    std::vector<std::pair<GlobalPoint, double>> tmp_seedPositionsEnergy;
+    if (seedingType_ == HistoMaxC3d) {
+      tmp_seedPositionsEnergy = computeMaxSeeds(smoothHistoCluster);
+      std::copy(tmp_seedPositionsEnergy.begin(), tmp_seedPositionsEnergy.end(), std::inserter(seedPositionsEnergy, seedPositionsEnergy.end()));
+    }
+    else if (seedingType_ == HistoThresholdC3d) {
+      tmp_seedPositionsEnergy = computeThresholdSeeds(smoothHistoCluster);
+      std::copy(tmp_seedPositionsEnergy.begin(), tmp_seedPositionsEnergy.end(), std::inserter(seedPositionsEnergy, seedPositionsEnergy.end()));
+    }
+    else if (seedingType_ == HistoInterpolatedMaxC3d) {
+      tmp_seedPositionsEnergy = computeInterpolatedMaxSeeds(smoothHistoCluster);
+      std::copy(tmp_seedPositionsEnergy.begin(), tmp_seedPositionsEnergy.end(), std::inserter(seedPositionsEnergy, seedPositionsEnergy.end()));
+    }
+    else if (seedingType_ == HistoSecondaryMaxC3d) {
+      tmp_seedPositionsEnergy = computeSecondaryMaxSeeds(smoothHistoCluster);
+      std::copy(tmp_seedPositionsEnergy.begin(), tmp_seedPositionsEnergy.end(), std::inserter(seedPositionsEnergy, seedPositionsEnergy.end()));
+    }
```

Comment:
The copy doesn't need to be repeated each time. It can be called only once after the if statements.
Also it would be better to use a `std::back_inserter` instead of `std::inserter`